search-commons/src/test/java/no/unit/nva/search/ResourceClientTest.java

```diff
@@ -442,6 +448,168 @@ void searchWithUriReturnsOpenSearchAwsResponse(URI uri, int expectedCount) throw
+                .build()
+                .withFilter()
+                .requiredStatus(PUBLISHED, PUBLISHED_METADATA)
+                .apply()
+                .doSearch(searchClient);
+
+        var pagedSearchResourceDto = response.toPagedResponse();
+        var documents = pagedSearchResourceDto.hits();
+
+        for (var document : documents) {
+            var actualParentId =
+                document
+                    .get("entityDescription")
+                    .get("reference")
+                    .get("publicationContext")
+                    .get("id")
+                    .asText();
+
+            assertThat(actualParentId, containsString(expectedParentIdSuffix));
+        }
+
+        assertThat(pagedSearchResourceDto.hits().size(), is(equalTo(expectedHits)));
+        assertThat(pagedSearchResourceDto.totalHits(), is(equalTo(expectedHits)));
+    }
+
+    @Test
+    void shouldFindChaptersOfBookAnthology() throws ApiGatewayException {
+        // Given that a parent document exists of type BookAnthology,
+        // and the parent document has two children of type AcademicChapter,
+        // when a query filters by partOf type BookAnthology,
+        // then two child documents should be returned,
+        // and the parent ID should be found in each child document.
+
+        var uri =
+            UriWrapper.fromUri("https://x.org/?instanceTypePartOf=BookAnthology&from=0&size=20")
+                .getUri();
+        var expectedHits = 2;
+        var expectedParentIdSuffix = "01905518408c-dba987f7-0d84-4519-a625-89605672afc8";
+
+        var response =
+            ResourceSearchQuery.builder()
+                .fromQueryParameters(queryToMapEntries(uri))
+                .withRequiredParameters(FROM, SIZE)
+                .withDockerHostUri(URI.create(container.getHttpHostAddress()))
+                .build()
+                .withFilter()
+                .requiredStatus(PUBLISHED, PUBLISHED_METADATA)
+                .apply()
+                .doSearch(searchClient);
+
+        var pagedSearchResourceDto = response.toPagedResponse();
+        var documents = pagedSearchResourceDto.hits();
+
+        for (var document : documents) {
+            var actualParentId =
+                document
+                    .get("entityDescription")
+                    .get("reference")
+                    .get("publicationContext")
+                    .get("id")
+                    .asText();
+
+            var actualInstanceType =
+                document
+                    .get("entityDescription")
+                    .get("reference")
+                    .get("publicationInstance")
+                    .get("type")
+                    .asText();
+
+            assertThat(actualParentId, containsString(expectedParentIdSuffix));
+            assertThat(actualInstanceType, is(equalTo("AcademicChapter")));
+        }
+
+        assertThat(pagedSearchResourceDto.hits().size(), is(equalTo(expectedHits)));
```

Comment:
is it 1 hit for parent or 2 hits for children?

Reply:
This finds all "children" (chapters), so in this case 2 hits (both belonging to the same parent)

Reply:
Yes. we have one parent with two children in the test data....